internal/provider/common.go

```diff
@@ -10,8 +10,8 @@ import (
 	"time"
 
```

Comment:
rename the file name from common to something more meaningful, split into multiple files if needed